data_checker/main.resource_policy_check.py

```diff
@@ -0,0 +1,134 @@
+    new_item_list = convert_old_ids_to_new(old_item_list, item_dict)
+    # list od item IDs from dspace7 which can READ Anonymous
+    item_ids_list=[]
+    # get total pages for search
+    # max page size for this request is 100
+    response = rest_proxy.get('discover/search/objects?sort=score,'
+                              'DESC&size=100&page=0&configuration=default'
+                              '&dsoType=ITEM&embed=thumbnail&embed=item%2Fthumbnail')
+    response_json = convert_response_to_json(response)
+    totalPages = objects = response_json['_embedded']['searchResult']['page']['totalPages']
+    # get result from each page
+    # we don't get items which are withdrawn or discoverable
+    for page in range(totalPages):
+        response = rest_proxy.get('discover/search/objects?sort=score,DESC&size=100&page=' +
+                                  str(page) +
+                                  '&configuration=default&'
+                                  'dsoType=ITEM&embed=thumbnail&embed=item%2Fthumbnail')
+        response_json = convert_response_to_json(response)
+        objects = response_json['_embedded']['searchResult']['_embedded']['objects']
+        # add each object to result list
+        for item in objects:
+            item_ids_list.append(item['_embedded']['indexableObject']['id'])
+
+    # compare expected items in dspace5 and got items from dspace7
+    # log items, which we cannot find
+    item_url = 'core/items'
+    for id_ in new_item_list:
+        if id_ in item_ids_list:
+            item_ids_list.remove(id_)
+        else:
+            # check if we really don't have access to item in Dspace7
+            response = do_api_get_one(item_url, id_)
+            if not response.ok:
+                _logger.error('Do not exist resource policy of anonymous READ for item: ' +
+                              id_ + ' in dspace7!')
+    #now in new_item_list are items whose resource_policy
+    # was not found in dspace5
+    # it could be because in dspace7 is using inheritance for resource policies
+    # check if you have access for these items in dspace5
+    # based on their handles or there was import error
+    item_lindat_url = 'https://lindat.mff.cuni.cz/repository/xmlui/handle/'
+    # load handle_json
+    handle_json = read_json(handle_json)
+    # create dict
+    handle_dict = {}
+    # handle has to be defined for item and item has to exist
+    handle_dict = {item_dict[handle['resource_id']]: handle['handle']
+                   for handle in handle_json if
+                   handle['resource_type_id'] == 2 and
+                   handle['resource_id'] in item_dict}
+    # do request to dspace5 for remaining items
+    for id_ in item_ids_list:
+        response = requests.get(item_lindat_url + handle_dict[id_])
+        if not response.ok:
+            _logger.error('Do not exist resource policy of '
+                          'anonymous READ for item: ' + id_ + ' in dspace5!')

```

Comment:
I think it would be good to add SOME kind of log. Perhaps
`_logger.info(f"checked {count} of items in DSpace5 succesfully")` of course, count would have to introduced.
perhaps also `_logger.debug(f"item with handle {handle_dict[id_]} sucessfully found in DSpace5")` after each request

Reply:
so about which counts do you want statistics?

Reply:
count of checked items. Count of problematic items. Ideally, count of items that are in 7 and not in 5 as one number and then the opposite (are in 5 and not in 7) as another number.